### PR TITLE
Add correlation to analysis

### PR DESCRIPTION
**Pull request by denkv**
This PR is for reviewing the changes in analysis.

### CODE REVIEW

File: analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java [hunk 1]
```diff
@@ -81,121 +89,149 @@ public void run() throws Exception {
...
+                    clusterDataset = dp.getInstancesDatasetForClustering();
+
+                    predictionDataset = dp.getInstancesDatasetForPrediction();
+
+                    correlationDataset = dp.getInstancesDatasetForCorrelation();
+
+                    mappings = dp.getMappings();
+
+                    ResIterator expURIs = experimentModel.listSubjectsWithProperty(parametersURI);
+                    List expUris = expURIs.toList();
```
Comment (MichaelRoeder) [inline analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java]: please add `<>` to generics.

File: analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java [hunk 2]
```diff
@@ -81,121 +89,149 @@ public void run() throws Exception {
...
+            LOGGER.info(SparqlQueries.getExperimentGraphQuery(expUri, null));
+            experimentModel = storage.sendConstructQuery(SparqlQueries.getExperimentGraphQuery(expUri, null));
+
+            Instances predictionDataset = null;
+            Instances clusterDataset = null;
+            Instances igDataset = null;
+            Instances correlationDataset = null;
+            Instances currentData = null;
+            DataProcessor dp = new DataProcessor();
+            DataProcessor dpForCurrent = new DataProcessor();
```
Comment (MichaelRoeder) [inline analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java]: Check whether needed.

File: analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java [hunk 3]
```diff
@@ -81,121 +89,149 @@ public void run() throws Exception {
...
+
+                // Remove existing analysis results for this pair of benchmark and system instance.
+                Resource resultset = updatedModel.listResourcesWithProperty(RDF.type, HOBBIT.AnalysisResultset).next();
+                storage.sendUpdateQuery(SparqlQueries.deleteAnalysisResults(resultset.getURI(), Constants.PUBLIC_RESULT_GRAPH_URI));
+
+                String sparqlUpdateQuery = null;
+                //TODO:: handle null exception for sparql queries
+                sparqlUpdateQuery = SparqlQueries.getUpdateQueryFromDiff(experimentModel,
+                        updatedModel,
+                        GRAPH_URI);
```
Comment (MichaelRoeder) [inline analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java]: Please make sure that the original model is replaced with `null` and the updated model is initialized as empty model.

File: analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java [hunk 4]
```diff
@@ -234,13 +270,16 @@ private void notifyQueue(){
     protected static class AnalysisModel {
```
Comment (MichaelRoeder) [inline analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java]: Please move the static classes into own files.

**Comment by denkv** [comment]
Most requests are done (except for moving classes).

**Pull request merged**